### PR TITLE
Add helper for creating MLOperandDescriptor to simplify steps

### PR DESCRIPTION
**Pull request by inexorabletash**
The only non-trivial bits are a few cases where in the previous steps the shape was calculated after the MLOperandDescriptor was created, but this would not be an observable change to script.

***
<a href="https://pr-preview.s3.amazonaws.com/inexorabletash/webnn/pull/737.html" title="Last updated on Jul 25, 2024, 7:53 PM UTC (6cba044)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/737/71dd440...inexorabletash:6cba044.html" title="Last updated on Jul 25, 2024, 7:53 PM UTC (6cba044)">Diff</a>

### CODE REVIEW

**Comment by inexorabletash** [comment]
Deleting lines is good! This should be pretty straightforward. @huningxin & @fdwr can you please review?

**Comment by inexorabletash** [comment]
This would also simplify #676 significantly.

**Review by huningxin — APPROVED** [review]
LGTM!

File: index.bs [hunk 1]
```diff
@@ -5879,11 +5858,10 @@ partial interface MLGraphBuilder {
...
     1. If |trueValue|'s [=MLOperand/dataType=] is not equal to |falseValue|'s [=MLOperand/dataType=], then [=exception/throw=] a {{TypeError}}.
-    1. Let |descriptor| be a new {{MLOperandDescriptor}}.
-    1. Set |descriptor|.{{MLOperandDescriptor/dataType}} to |trueValue|'s [=MLOperand/dataType=].
-    1. Set |descriptor|.{{MLOperandDescriptor/dimensions}} to the result of [=bidirectionally broadcasting the shapes=] |trueValue|'s [=MLOperand/shape=] and |falseValue|'s [=MLOperand/shape=].
+    1. Let |outputShape| be the result of [=bidirectionally broadcasting the shapes=] |trueValue|'s [=MLOperand/shape=] and |falseValue|'s [=MLOperand/shape=].
         1. If that returns failure, then [=exception/throw=] a {{TypeError}}.
-    1. If |condition| is not [=bidirectionally broadcastable=] to |descriptor|.{{MLOperandDescriptor/dimensions}}, then [=exception/throw=] a {{TypeError}}.
+    1. If |condition| is not [=bidirectionally broadcastable=] to |outputShape|, then [=exception/throw=] a {{TypeError}}.
+    1. Let |descriptor| be the result of [=creating an MLOperandDescriptor=] given |trueValue|'s [=MLOperand/dataType=] and |outputShape|.
```
Comment (fdwr) [inline index.bs:5864]:
```suggestion
    1. If |condition| is not [=bidirectionally broadcastable=] to |outputShape|, then [=exception/throw=] a {{TypeError}}.
    1. Let |outputShape| be the result of [=bidirectionally broadcasting the shapes=] |condition|'s [=MLOperand/shape=] and |outputShape].
    1. Let |descriptor| be the result of [=creating an MLOperandDescriptor=] given |trueValue|'s [=MLOperand/dataType=] and |outputShape|.
```
Looks like we're missing a step here. We *check* that `condition` is bidirectionally broadcastable with the intermediate `outputShape` of trueValue and falseValue, but we don't actually update `outputShape` accordingly.

**Review by fdwr — APPROVED** [review]
This should cut down on the tedium some. Looks good with one addition.

File: index.bs (hunk 1 above)
Comment (inexorabletash) [reply index.bs:5864]: Just to be clear, this was a pre-existing problem, right?

Comment (inexorabletash) [reply index.bs]: Updated in c938542 too.

Comment (fdwr) [reply index.bs]: > Just to be clear, this was a pre-existing problem, right?

Yep, existing.

File: index.bs [hunk 2]
```diff
@@ -5860,7 +5860,8 @@ partial interface MLGraphBuilder {
     1. If |trueValue|'s [=MLOperand/dataType=] is not equal to |falseValue|'s [=MLOperand/dataType=], then [=exception/throw=] a {{TypeError}}.
     1. Let |outputShape| be the result of [=bidirectionally broadcasting the shapes=] |trueValue|'s [=MLOperand/shape=] and |falseValue|'s [=MLOperand/shape=].
         1. If that returns failure, then [=exception/throw=] a {{TypeError}}.
-    1. If |condition| is not [=bidirectionally broadcastable=] to |outputShape|, then [=exception/throw=] a {{TypeError}}.
+    1. Set |outputShape| be the result of [=bidirectionally broadcasting the shapes=] |condition|'s [=MLOperand/shape=] and |outputShape].
```
Comment (fdwr) [inline index.bs:5863]:
```suggestion
    1. Set |outputShape| to the result of [=bidirectionally broadcasting the shapes=] |condition|'s [=MLOperand/shape=] and |outputShape].
```

**Review by fdwr — APPROVED** [review]
👍 TY. Fixing a small typo and merging.

**Pull request merged**